File: tests/src/org.apache.hadoop.thirdparty/hadoop-shaded-protobuf_3_7/1.1.1/src/test/java/org_apache_hadoop_thirdparty/hadoop_shaded_protobuf_3_7/ByteBufferWriterTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_hadoop_thirdparty.hadoop_shaded_protobuf_3_7;

import static org.assertj.core.api.Assertions.assertThat;

import org.apache.hadoop.thirdparty.protobuf.ByteString;
import org.junit.jupiter.api.Test;

public class ByteBufferWriterTest {
    private static final String BYTE_BUFFER_WRITER_CLASS_NAME =
            "org.apache.hadoop.thirdparty.protobuf.ByteBufferWriter";

    @Test
    void classInitializationDiscoversFileOutputStreamChannelField() throws Exception {
        Class<?> writerClass = Class.forName(
                BYTE_BUFFER_WRITER_CLASS_NAME,
                true,
                ByteString.class.getClassLoader()
        );

        assertThat(writerClass.getName()).isEqualTo(BYTE_BUFFER_WRITER_CLASS_NAME);
    }
}

File: tests/src/com.typesafe.akka/akka-protobuf-v3_2.13/2.8.8/src/test/scala/com_typesafe_akka/akka_protobuf_v3_2_13/ByteBufferWriterTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_typesafe_akka.akka_protobuf_v3_2_13

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

import akka.protobufv3.internal.ByteString

class ByteBufferWriterTest {
  @Test
  def classInitializationDiscoversFileOutputStreamChannelField(): Unit = {
    val writerClass: Class[_] = Class.forName(
      "akka.protobufv3.internal.ByteBufferWriter",
      true,
      classOf[ByteString].getClassLoader
    )

    assertEquals("akka.protobufv3.internal.ByteBufferWriter", writerClass.getName)
  }
}
